receiver/podmanreceiver/podman.go

```diff
@@ -0,0 +1,206 @@
+	defer pc.containersLock.Unlock()
+	containers := make([]Container, 0, len(pc.containers))
+	for _, container := range pc.containers {
+		containers = append(containers, container)
+	}
+	return containers
+}
+
+// LoadContainerList will load the initial running container maps for
+// inspection and establishing which containers warrant stat gathering calls
+// by the receiver.
+func (pc *ContainerScraper) LoadContainerList(ctx context.Context) error {
+	params := url.Values{}
+	runningFilter := map[string][]string{
+		"status": {"running"},
+	}
+	jsonFilter, err := json.Marshal(runningFilter)
+	if err != nil {
+		return nil
+	}
+	params.Add("filters", string(jsonFilter[:]))
+
+	listCtx, cancel := context.WithTimeout(ctx, pc.config.Timeout)
+	defer cancel()
+	containerList, err := pc.client.list(listCtx, params)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range containerList {
+		pc.persistContainer(c)
+	}
+	return nil
+}
+
+func (pc *ContainerScraper) Events(ctx context.Context, options url.Values) (<-chan Event, <-chan error) {
+	return pc.client.events(ctx, options)
+}
+
+func (pc *ContainerScraper) ContainerEventLoop(ctx context.Context) {
+	filters := url.Values{}
+	cidFilter := map[string][]string{
+		"status": {"died", "start"},
+		"type":   {"container"},
+	}
+	jsonFilter, err := json.Marshal(cidFilter)
+	if err != nil {
+		return
+	}
+	filters.Add("filters", string(jsonFilter[:]))
+EVENT_LOOP:
+	for {
+		eventCh, errCh := pc.Events(ctx, filters)
+		for {
+
+			select {
+			case <-ctx.Done():
+				return
+			case event := <-eventCh:
+				switch event.Status {
+				case "died":
+					pc.logger.Debug("Podman container died:", zap.String("id", event.ID))
+					pc.removeContainer(event.ID)
+				case "start":
+					pc.logger.Debug(
+						"Podman container started:",
+						zap.String("id", event.ID),
+						zap.String("status", event.Status),
+					)
+					pc.InspectAndPersistContainer(ctx, event.ID)
+				}
+			case err := <-errCh:
+				// We are only interested when the context hasn't been canceled since requests made
+				// with a closed context are guaranteed to fail.
+				if ctx.Err() == nil {
+					pc.logger.Error("Error watching podman container events", zap.Error(err))
+					// Either decoding or connection error has occurred, so we should resume the event loop after
+					// waiting a moment.  In cases of extended daemon unavailability this will retry until
+					// collector teardown or background context is closed.
+					select {
+					case <-time.After(3 * time.Second):
+						continue EVENT_LOOP
+					case <-ctx.Done():
+						return
+					}
+				}
+			}
+
+		}
+	}
+}
+
+// InspectAndPersistContainer queries inspect api and returns *ContainerStats and true when container should be queried for stats,
+// nil and false otherwise. Persists the container in the cache if container is
+// running and not excluded.
+func (pc *ContainerScraper) InspectAndPersistContainer(ctx context.Context, cid string) (*Container, bool) {
```

Comment:
This function returns the container's details and whether or not it is persisted, but I don't see any uses of its return values.  Is it only for testing?

Reply:
Exactly, indeed it is returning the container information to maintain the format with the docker implementation: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/docker/docker.go#L280